File: Compiler/src/main/kotlin/nodes/IdentListNode.java
package nodes;

public class IdentListNode {
    IdentNode first;
    IdentNode last;

    public IdentListNode(String ID) {
        IdentNode newIndentNode = new IdentNode(ID);
        this.first = newIndentNode;
        this.last = newIndentNode;
    }

    public void addIdent(String ID) {
        IdentNode newIndentNode = new IdentNode(ID);
        this.last.next = newIndentNode;
        this.last = newIndentNode;
    }
}

File: Compiler/src/main/kotlin/nodes/IdentListNode.kt
package nodes

class IdentListNode(ID: String?) {
    var first: IdentNode
    var last: IdentNode

    init {
        val newIndentNode = IdentNode(ID)
        first = newIndentNode
        last = newIndentNode
    }

    fun addIdent(ID: String?) {
        val newIndentNode = IdentNode(ID)
        last.next = newIndentNode
        last = newIndentNode
    }
}